src/metadata.jl

```diff
@@ -243,4 +245,5 @@ Base.eltype(::Metadata{T}) where T = T
 fill_value_decoding(v::AbstractString, T::Type{<:Number}) = parse(T, v)
 fill_value_decoding(v::Nothing, ::Any) = v
 fill_value_decoding(v, T) = T(v)
+fill_value_decoding(v::Number, T::Type{String}) = v == 0 ? "" : T(UInt8[v])
```

Comment:
Could you quickly reference an example metadata file where fill values for string arrays are encoded as numbers? I have problems to understand what is going on here.

Reply:
One way to create something like that is to do the following:
```
import zarr
import numpy as np

test = zarr.open("test.zarr")
test2 = np.asarray(["test1", "test2"], dtype="O")
test.create_dataset("test2", shape=test2.shape, dtype=str)
test["test2"][:] = test2
```
The `.zarray` then has the following contents:
```
{
    "chunks": [
        2
    ],
    "compressor": {
        "blocksize": 0,
        "clevel": 5,
        "cname": "lz4",
        "id": "blosc",
        "shuffle": 1
    },
    "dtype": "|O",
    "fill_value": 0,
    "filters": [
        {
            "id": "vlen-utf8"
        }
    ],
    "order": "C",
    "shape": [
        2
    ],
    "zarr_format": 2
}
```

Note that this is what [AnnData is doing to store string arrays.](https://github.com/scverse/anndata/blob/main/anndata/_io/specs/methods.py#L370)

Reply:
Ok that would explain the first branch but I think `T(UInt8[v])` would error in all cases, because you can not index the type `UInt8`. Shall we error for non-zero numbers or just do `T(v)`?

Reply:
Maybe we could add you small example in the python.jl part of the unit tests where we look at this type of interop?

Reply:
`UInt8[v]` creates a byte array from `v`, the assumption being that any number used as fill value is a byte sequence of a string. On my system, this runs fine:
```
julia> String(UInt8[247])
"\xf7"
```

Reply:
Of course, probably I should not be as tired as yesterday when making PR reviews.